### PR TITLE
Updating feature/rrfs-physics with changes from feature/hafs.v2.0.1 branch

### PR DESCRIPTION
**Pull request by mrinalbiswas**
## Description of changes
Provide a description of what this PR does. What bug does it fix, or what feature does it add? Do you expect that this PR will change answers, and if so, under what circumstances? If this PR is for a physics innovation, please provide references to any relevant scientific papers.

## Issues addressed (optional)
If this PR addresses one or more issues, please provide link(s) to the issue(s) here.
- fixes hafs-community/HAFS/issues/<issue_number>

## Dependencies (optional)
If submodule PRs are required, please link them below. For example:
- hafs-community/ufs-weather-model/pull/<pr_number>
- hafs-community/UPP/pull/<pr_number>
- hafs-community/UFS_UTILS/pull/<pr_number>
- hafs-community/GSI/pull/<pr_number>

## Contributors (optional)
If others worked on this PR besides the author, please include their user names here (using @Mention if possible).

## Tests conducted
What testing has been conducted on the PR thus far? Describe the nature of any scientific or technical tests, including relevant details about the configuration(s) (e.g., cold versus warm start, number of cycles, forecast length, whether data assimilation was performed, etc). What platform(s) were used for testing?

## Application-level regression test status
Running the HAFS application-level regression tests is currently performed by code reviewers after the developer creates the initial PR. As regression tests are conducted, the testers should use the checklist below to indicate **successful** regression tests. You may add other tests as needed. If a test fails, do not check the box. Instead, describe the failure in the PR comments, noting the platform where the test failed.

- [ ] Jet
- [ ] Hera
- [ ] Orion
- [ ] WCOSS2

### CODE REVIEW

File: modulefiles/hafs.hercules.lua [hunk 1]
```diff
@@ -40,7 +40,7 @@ load(pathJoin("netcdf-fortran", netcdf_fortran_ver))
 parallelio_ver=os.getenv("parallelio_ver") or "2.5.10"
 load(pathJoin("parallelio", parallelio_ver))
 
-esmf_ver=os.getenv("esmf_ver") or "8.5.0"
+esmf_ver=os.getenv("esmf_ver") or "8.6.0"
```
Comment (SamuelTrahanNOAA) [inline modulefiles/hafs.hercules.lua:43]: This conflicts with the version of esmf used inside the ufs-weather-model. That could cause errors at runtime.

If you can confirm at least one cycle of the coupled model runs with this change, we can assume it's safe on that machine.

File: jobs/JHAFS_OCN_PREP [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 date
 export PS4='+ $SECONDS + '
-set -xue
+set -x
```
Comment (SamuelTrahanNOAA) [inline jobs/JHAFS_OCN_PREP:5]: Turning off the "u" and "e" eliminates error checking. You probably need to turn them off during the module commands, but not for the whole script.

I suggest you retain the "set -xue" here, but:

1. Add `set +ue` before the series of "if" statements with module commands
2. Add `set -ue` after the series of "if" statements with module commands

**Review by SamuelTrahanNOAA — COMMENTED** [review]
It looks good overall, but I found two problems you should look into.

File: modulefiles/hafs.hercules.lua (hunk 1 above)
Comment (mrinalbiswas) [reply modulefiles/hafs.hercules.lua:43]: @SamuelTrahanNOAA I can confirm that at least two cycles with 8.6.0.

File: jobs/JHAFS_OCN_PREP (hunk 2 above)
Comment (mrinalbiswas) [reply jobs/JHAFS_OCN_PREP:5]: @SamuelTrahanNOAA The feature/hafs.v2.0.1 contains just set -x. 
@BinLiu-NOAA removed "euo".

commit c27e49d2030b8c8403b926f096dfc4c4f107e0c1
Author: Bin Liu <Bin.Liu@noaa.gov>
Date:   Thu Apr 18 19:17:13 2024 +0000

    Take out "set +xeuo" lines and improve the "set -" settings for the
    jobs, scripts, ush scripts (based on NCO/SPA's and EE2 Reviewer's suggestions)

**Review by SamuelTrahanNOAA — APPROVED** [review]

**Pull request merged**